### PR TITLE
fix: textArea component maxlength slice garbled

### LINKED ISSUE #27612: textarea最大长度截取问题
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/ant-design/ant-design/blob/f5ed19248f95a4561f9c33800756228d4efed3ce/components/input/TextArea.tsx#L116](https://github.com/ant-design/ant-design/blob/f5ed19248f95a4561f9c33800756228d4efed3ce/components/input/TextArea.tsx#L116)

### Steps to reproduce
传入一个😂 ，maxLength设为1，传入就出现乱码

### What is expected?
改变截取方式

### What is actually happening?
  

| Environment | Info |
|---|---|
| antd | 4.8.0 |
| React | 计算长度用的[...value].length,但是截取直接用的value.slice(0,maxLength)) |
| System | all |
| Browser | all |

### PR DESCRIPTION
**Pull request by Nuomic**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #27612

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
Could you add test case and fill the PR template content?

File: components/input/TextArea.tsx [hunk 1]
```diff
@@ -113,7 +113,7 @@ class TextArea extends React.Component<TextAreaProps, TextAreaState> {
 
     // Max length value
     const hasMaxLength = Number(maxLength) > 0;
-    value = hasMaxLength ? value.slice(0, maxLength) : value;
+    value = hasMaxLength ? Array.from(value).slice(0, maxLength).join('') : value;
```
Comment (afc163) [inline components/input/TextArea.tsx:116]: Array.from 在 IE 下不兼容。

Comment (zhangweiqiang666) [reply components/input/TextArea.tsx:116]: ![Uploading 8]YVL1HTZVTBE}%D2FZ_(YB.png…]()

Comment (zhangweiqiang666) [reply components/input/TextArea.tsx:116]: > Array.from 在 IE 下不兼容。

感谢提醒，但同时  [...value].length 在 iE下 经过babel编译也有不同的表现。
8 YVL1HTZVTBE}%D2FZ_(YB

**Pull request closed without merging**